internal/app/cache/cache.go

```diff
@@ -185,3 +204,16 @@ func marshalResources(resources []*any.Any) ([]gcp_types.MarshaledResource, erro
 	}
 	return marshaledResources, nil
 }
+
+func removeRequests(request *v2.DiscoveryRequest, requests []*v2.DiscoveryRequest) []*v2.DiscoveryRequest {
+	requestsAfterDeletion := make([]*v2.DiscoveryRequest, len(requests))
+	i := 0
+	for _, r := range requests {
+		if r != request {
+			requestsAfterDeletion[i] = r
+			i++
+		}
```

Comment:
I'm thinking this loop could be very long. Wondering if there was a different way.

Reply:
I thought about using a map to track all the requests instead, but there's still a tradeoff in efficiency. Using a list like we do today makes AddRequest and SetResponse more efficient and DeleteRequest less efficient w.r.t requests. Using a map would be AddRequest and DeleteRequest more efficient but SetResponse less efficient (need to iterate over the map to get the keys), so it depends on whether we expect to call SetResponse or DeleteRequest more often.

Reply:
I think the frequency of Add/Delete Requests is much more than SetResponse.
The scenario i'm thinking about is, say, during scaleup 1000 pods came up. When they get removed during scaledown 1000 sidecar DeleteRequest will cause 1000^2 iterations.

Reply:
As discussed, since the caller of SetResponse would iterate over the return value (list of requests) anyway, it should make no difference to return a map of requests instead for the caller to iterate over.

Reply:
The map value type could also be an empty struct if the added memory from carrying around an extra bool is too much, but that might be excessively optimizing.